### PR TITLE
Various additions for QtWebEngine and Chromium

### PR DESCRIPTION
**Pull request by Dennisbonke**
This PR aims to implement missing functions, defines and structs for `QtWebEngine` and `Chromium`. As my first aim was `QtWebEngine`, the branch is named after that, but the current focus really is `Chromium`. At this moment in time, chromium does compile and link, but I expect more additions to be needed that we'll find while running and debugging it. To avoid making this too big, I'd like to get this merged in as is and do the rest in a separate PR.

Related PRs can be found in #458 and #461

### CODE REVIEW

**Comment by avdgrinten** [comment]
This PR is getting huge. Can it be split up into smaller chunks that can be merged more quickly and iteratively? In general, we should avoid large "add everything required for X" PRs.

**Comment by Dennisbonke** [comment]
> This PR is getting huge. Can it be split up into smaller chunks that can be merged more quickly and iteratively? In general, we should avoid large "add everything required for X" PRs.

I agree. I likely have some work laying around uncommitted but after that this PR should be merged and further work will be split off.

**Comment by Dennisbonke** [comment]
No additional local changes were found, ready for review. Expect tests for `timeradd()`, `timersub()`, `timerclear()` and `timerisset()` soon.

**Comment by Dennisbonke** [comment]
Rebased and dropped changes merged in by #429

File: options/ansi/generic/string-stubs.cpp [hunk 1]
```diff
@@ -485,3 +485,17 @@ void *memmem(const void *, size_t, const void *, size_t) {
 	__ensure(!"Not implemented");
 	__builtin_unreachable();
 }
+
+// BSD extension.
+// Taken from musl
+size_t strlcpy(char *d, const char *s, size_t n) {
```
Comment (Geertiebear) [inline options/ansi/generic/string-stubs.cpp:491]: Please add a test.

File: options/posix/include/bits/posix/posix_signal.h [hunk 2]
```diff
@@ -11,6 +11,51 @@ extern "C" {
...
+#define FPE_FLTUND      5       /* floating point underflow */
+#define FPE_FLTRES      6       /* floating point inexact result */
+#define FPE_FLTINV      7       /* floating point invalid operation */
+#define FPE_FLTSUB      8       /* subscript out of range */
+
+#define TRAP_BRKPT      1       /* process breakpoint */
+#define TRAP_TRACE      2       /* process trace trap */
+
+// Start Glibc stuff
+#define __ctx(fld) fld
```
Comment (Geertiebear) [inline options/posix/include/bits/posix/posix_signal.h:29]: What's the point of this macro?

Comment (Dennisbonke) [reply options/posix/include/bits/posix/posix_signal.h]: It seems unused, removed it.

File: options/ansi/generic/string-stubs.cpp (hunk 1 above)
Comment (Dennisbonke) [reply options/ansi/generic/string-stubs.cpp:491]: Done, also added `strlcat` and tests for that as well.

**Review by Geertiebear — APPROVED** [review]
LGTM.

**Pull request merged**